src/glcm.jl

```diff
@@ -0,0 +1,150 @@
+function glcm{T<:Real}(img::AbstractArray{T, 2}, distance::Integer, angle::Real, mat_size::Integer = 16)
+    max_img = maximum(img)
+    img_rescaled = map(i -> max(1, Int(ceil(i * mat_size / max_img))), img)
+    _glcm(img_rescaled, distance, angle, mat_size)
+end
+
+function glcm{T<:Real, A<:Real}(img::AbstractArray{T, 2}, distances::Array{Int, 1}, angles::Array{A, 1}, mat_size::Integer = 16)
+    max_img = maximum(img)
+    img_rescaled = map(i -> max(1, Int(ceil(i * mat_size / max_img))), img)
+    glcm_matrices = [_glcm(img_rescaled, d, a, mat_size) for d in distances, a in angles]
+    glcm_matrices
+end
+
+function glcm{T<:Colorant}(img::AbstractArray{T, 2}, distance::Integer, angle::Real, mat_size::Integer = 16)
+    img_rescaled = map(i -> max(1, Int(ceil((convert(Gray{U8}, i).val.i) * mat_size / 256))), img)
+    _glcm(img_rescaled, distance, angle, mat_size)
+end
+
+function glcm{T<:Colorant, A<:Real}(img::AbstractArray{T, 2}, distances::Array{Int, 1}, angles::Array{A, 1}, mat_size::Integer = 16)
+    img_rescaled = map(i -> max(1, Int(ceil((convert(Gray{U8}, i).val.i) * mat_size / 256))), img)  
+    glcm_matrices = [_glcm(img_rescaled, d, a, mat_size) for d in distances, a in angles]
+    glcm_matrices
+end
+
+glcm{T<:Union{Colorant, Real}, A<:Real}(img::AbstractArray{T, 2}, distances::Int, angles::Array{A, 1}, mat_size::Integer = 16) = glcm(img, [distances], angles, mat_size)
+glcm{T<:Union{Colorant, Real}}(img::AbstractArray{T, 2}, distances::Array{Int, 1}, angles::Real, mat_size::Integer = 16) = glcm(img, distances, [angles], mat_size)
+
+function _glcm{T}(img::AbstractArray{T, 2}, distance::Integer, angle::Number, mat_size::Integer)
+    co_oc_matrix = zeros(Integer, mat_size, mat_size)
+    for j = 1:size(img, 1), i = 1:size(img, 2)
+        int_one = img[j, i]
+        j_off = j + Int(round(sin(angle) * distance))
```

Comment:
precompute `sin(angle)` and `cos(angle)` and store them as constants---they're fairly slow, and since they are const there is no point recomputing them on each iteration of the loop.

If you really want to get fancy, you can even break this into two loops and do the bounds-checking on `j` and perhaps skip the entire loop over `i`...